File: src/LCP66.600YaG.1000481/solution.ts
function minNumBooths (demand: string[]): number {
    const result = new Array<number>(26).fill(0);
    for (let i = 0; i < demand.length; i++) {
        const day = countDay(demand[i]);
        for (let i = 0; i < 26; i++) {
            result[i] = Math.max(result[i], day[i]);
        }
    }
    return result.reduce((sum, item) => sum + item, 0);
}

function countDay (str:string) {
    const result = new Array<number>(26).fill(0);
    for (let i = 0; i < str.length; i++) {
        result[str.charCodeAt(i) - 97]++;
    }
    return result;
}

File: src/LCP66.600YaG.1000481/solution.java
class Solution {
    public int minNumBooths(String[] demand) {
        int[] total = new int[26];
        for(String d:demand){
            int[] counts = calc(d.toCharArray());
            for(int i=0;i<26;i++){
                total[i] = Math.max(total[i],counts[i]);
            }
        }
        int result = 0;
        for(int i=0;i<26;i++){
            result += total[i];
        }
        return result;
    }
    
    private int[] calc(char[] chars){
        int[] counts = new int[26];
        for(int c:chars){
            counts[c-'a']++;
        }
        return counts;
    }
}